File: 0011-container-with-most-water/0011-container-with-most-water.kt
class Solution {
    fun maxArea(height: IntArray): Int {
       val n =  height.size
       var r = n-1
       var l = 0
       var max_area = 0
       while(r>l){
            val h = min(height[r],height[l])
            val w = r-l
            val area = w*h
            max_area = max(max_area,area)

            if(height[r]<height[l]){
                r--;
            }else{
                l++;
            }
       }
       return max_area;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n= height.size();
        int l =0, r=n-1;
        int max_area = 0;
        while(l<r){
            int w = r-l;
            int h = min(height[r],height[l]);
            int area = w*h;
            max_area = max(max_area,area);
            if(height[r]<height[l]){
                r--;
            }else{
                l++;
            }
        }
        return max_area;
    }
};